src/t_hash.c

```diff
@@ -3117,14 +3139,38 @@ static void hexpireGenericCommand(client *c, const char *cmd, long long basetime
     hashTypeSetExInit(keyArg, hashObj, c, c->db, cmd, expireSetCond, &exCtx);
     addReplyArrayLen(c, numFields);
 
-    for (int i = 0 ; i < numFields ; i++) {
-        sds field = c->argv[numFieldsAt+i+1]->ptr;
+    fieldAt = numFieldsAt + 1;
+    while (fieldAt < c->argc) {
+        sds field = c->argv[fieldAt]->ptr;
         SetExRes res = hashTypeSetEx(hashObj, field, expire, &exCtx);
+
+        if (unlikely(res != HSETEX_OK)) {
+            /* If the field was not set, prevent field propagation */
+            rewriteClientCommandArgument(c, fieldAt, NULL);
+            fieldsNotSet = 1;
+        } else {
```

Comment:
if the command is `hexpire h 9999 FIELDS 5 a <noset1> b <noset2> c`, after rewriting, c->argv will become `hexpire h 9999 FIELDS 3 a NULL b NULL c`, which seems werid.

Reply:
it shifts left all the arguments of the command as well. Please look inside function `rewriteClientCommandArgument`. I added there code (that should be reviewed as well) :
```
        /* move the remaining arguments one step left */
        for (int j = i+1; j < c->argc; j++) {
            c->argv[j-1] = c->argv[j];
        }
```
In case you concern about performance if there are many arguments, please take into consideration that it is not a common case to drop many arguments because they don't exists, condition not met, or their time is in the past.

Reply:
This loop is a bit hard to understand as `rewriteClientCommandArgument()` updates 
`c->argc` & `c->argv`. I guess we don't have any better way.